test/tests/functional/pbs_qrun.py

```diff
@@ -92,3 +93,39 @@ def test_invalid_host_val(self):
         msg = "Server is not up"
         self.assertTrue(self.server.isUp(), msg)
         self.logger.info("As expected server is up and running")
+
+    def test_qrun_hangs(self):
+        """
+        This test verifies whether qrun hangs, if scheduling is
+        disabled
+        """
+        node = self.mom.shortname
+        time1 = int(time.time())
+        while True:
+            self.server.manager(MGR_CMD_SET, NODE,
+                                {'resources_available.ncpus': 1}, id=node)
+            j1 = Job(TEST_USER)
+            j1.create_script('/bin/date')
+            self.server.submit(j1)
+            jid2 = self.server.submit(j1)
+
+            self.server.manager(MGR_CMD_SET, SCHED,
+                                {'scheduling': 'True'})
+            self.server.manager(MGR_CMD_SET, SCHED,
+                                {'scheduling': 'False'})
+            self.server.manager(MGR_CMD_SET, NODE,
+                                {'resources_available.ncpus': 2}, id=node)
+            self.server.runjob(jobid=jid2)
+
+            try:
+                self.scheduler.log_match("Starting Scheduling Cycle",
+                                         starttime=self.server.ctime,
+                                         max_attempts=10)
+            except PtlLogMatchError:
+                self.assertFalse(True,
+                                 "New scheduling cycle not started.Qrun hangs")
+            else:
+                time2 = int(time.time())
+                if (time2 - time1) > 60:
+                    self.logger.info("Loop ran since 1 min,no hang noticed")
+                    break
```

Comment:
I don't think this will work.  First, the runjob will call qrun.  Won't that hang?  Also, you set scheduling to true at the top.  This should run a cycle regardless of qrun.  The log_match() should match that every time since you set starttime=self.server.ctime

Reply:
starttime=self.server.ctime => Used this to indicate the log_match should start matching from the current server time. Is there any other variable/parameter I should use?

Regarding the "scheduling", at first it set to true to get a new cycle and then set to false, as the script expect qrun to trigger the scheduling cycle.

Reply:
self.server.ctime is the time that the server object was created.  If I understand this correctly, this should be the time the test suite started.  This means any scheduling cycle that has started since you ran bench_press will match.